### PR TITLE
Unified 'Reopen' action (#377)

### PR DESCRIPTION
**Pull request by Sir-Thomas**
This change replaces the current "Mark as Not Done" and "Undismiss" options when right clicking a closed task with a single "Reopen" option as suggested in #377.

This is my first time contributing to an open source project, so let me know if there is anything that I need to change.

### CODE REVIEW

**Comment by diegogangl** [comment]
Hey welcome and thanks!
Code looks ok, and works great. The only thing to change would be the commit message. Try using a verb (like "Add unified reopen action"). The number should be in the description part of the message (leave a blank line after the first line). Also you can use keywords to automatically close issues in Github, so you could just write  `fix #377` to autoclose that issue once this is merged in master.

More info: https://chris.beams.io/posts/git-commit/

File: GTG/gtk/browser/main_window.py [hunk 1]
```diff
@@ -1315,6 +1315,26 @@ def on_dismiss_task(self, widget=None):
                 task.set_status(Task.STA_DISMISSED)
                 self.close_all_task_editors(uid)
 
+    def on_reopen_task(self, widget=None):
+        tasks_uid = [uid for uid in self.get_selected_tasks()
+                     if uid is not None]
+        if len(tasks_uid) == 0:
+            return
```
Comment (Neui) [inline GTG/gtk/browser/main_window.py:1322]: Not needed. `tasks_uid` is an emtpy list, so the following `tasks` and `tasks_status` are empty as well, and `zip([], [], [])` is empty too, so the loop isn't entered.
I know you've copied from above, but if you're already here you can change this instance.

File: GTG/gtk/browser/main_window.py [hunk 2]
```diff
@@ -1315,6 +1315,26 @@ def on_dismiss_task(self, widget=None):
...
+        tasks = [self.req.get_task(uid) for uid in tasks_uid]
+        tasks_status = [task.get_status() for task in tasks]
+        for uid, task, status in zip(tasks_uid, tasks, tasks_status):
+            if status == Task.STA_DONE:
+                # Marking as undone
```
Comment (Neui) [inline GTG/gtk/browser/main_window.py:1327]: This is already clear from the context.  I know you've copied from above, but if you're already here you can change this instance.

**Review by Neui — APPROVED** [review]
LGTM and works. Suggesting removing some unnecessary lines. Maybe also put the shortcut in the shortcuts window.

**Comment by Sir-Thomas** [comment]
I have made the changes you guys suggested. Do you want to take another look at it and see if I understood everything correctly?

**Comment by diegogangl** [comment]
LGTM, one last thing for me:

Could you squash commit `40fa626ac6d3015d741fdd4c69aa0726caae44d7` onto `c9737dbfc0a62361b2111878c7a3a3ace2f54237` to make the history cleaner?

**Comment by diegogangl** [comment]
Hi, @Sir-Thomas did you have a chance to could squash the commits? It'd be great to land this for 0.6

**Comment by diegogangl** [comment]
No problem, thanks!

**Pull request merged**